File: hrmsFe/src/app/model/updateAdvertisementResponse.ts
import {GetPersonalSkillResponse} from "./getPersonalSkillResponse";
import {GetHumanResourceEntityResponse} from "./getHumanResourceEntityResponse";

export interface UpdateAdvertisementResponse{
  id:string,

  advCode:string,

  advertisementStatue:string,

  title:string,

  activationTime:Date,

  offDate:Date,

  jobDescription:string,

  personalSkills: GetPersonalSkillResponse[],

  humanResourceEntity:GetHumanResourceEntityResponse
}

File: hrms/src/main/java/com/obss/hrms/response/UpdateAdvertisementResponse.java
package com.obss.hrms.response;

import com.obss.hrms.entity.AdvertisementStatue;
import com.obss.hrms.entity.HumanResourceEntity;
import com.obss.hrms.entity.PersonalSkill;

import java.time.LocalDate;
import java.util.List;

public record UpdateAdvertisementResponse(
        String id,
        String advCode,
        String title,
        AdvertisementStatue advertisementStatue,
        LocalDate activationTime,
        LocalDate offDate,
        String jobDescription,
        List<PersonalSkill> personalSkills,
        HumanResourceEntity humanResourceEntity

) {
}
